array.h

```diff
@@ -56,6 +69,30 @@
 // functions have internal linkage.
 #define NDARRAY_UNIQUE static
 
+// Functions attributed with NDARRAY_HOST_DEVICE can run on both device and host mode on CUDA.
+#if defined(__clang__) && defined(__CUDA__) && !defined(__CUDA_ARCH__)
+// clang compiling CUDA code, host mode.
+#define NDARRAY_HOST_DEVICE
+#elif defined(__clang__) && defined(__CUDA__) && defined(__CUDA_ARCH__)
```

Comment:
Maybe a nested #if would be better here?

Reply:
Moot.